### PR TITLE
feat(plugin): replace old bus implementation using the new `x-priority-bus`

### PR DESCRIPTION
**Pull request by LuisMartinez15**
Replaced the old bus implementation with the new `@empathyco/x-bus`.

### CODE REVIEW

File: packages/x-components/src/plugins/x-bus.ts [hunk 1]
```diff
@@ -1,82 +1,24 @@
...
-      this.emitters[event] ??
-      (this.emitters[event] = new ReplaySubject<WirePayload<XEventPayload<Event>>>(1) as any)
-    );
-  }
-}
-
-/** @internal The bus instance. Will be replaced by injection. */
-export const bus: XBus = new BaseXBus();
-// TODO Remove this instantiation and replace with injection where used
+export const bus = new XPriorityBus<XEventsTypes, WireMetadata>({
```
Comment (annacv) [inline packages/x-components/src/plugins/x-bus.ts:11]: Would be nice to specify the direction of the priorities to have it documented

**Comment by annacv** [comment]
I ran the x-components test in local and there is one that failed, but now the error in jenkins is another one :( 
Here is the test that failed in local
Captura de pantalla 2023-02-28 a les 15 24 46

**Comment by mavmaf** [comment]
Thank God I don't have to review 94 files 😅 
There is something going with the scrolling when a filter is selected. Please check the video attached

https://user-images.githubusercontent.com/77147901/222127741-9109a65d-98c8-404e-bbb4-1e1b1f7c2f11.mov

**Comment by LuisMartinez15** [comment]
> Thank God I don't have to review 94 files 😅 There is something going with the scrolling when a filter is selected. Please check the video attached: firstly, without any filter selected the behaviour is as expected; secondly, the scroll is not kept

The scroll is kept, the problem is that the page is not. We'll probably need to check events execution order & priorities in order to fix this issue. Thanks for the report!

**Review by mnavarroespinosa — COMMENTED** [review]
I didn't find any issue apart from the one that Mavi reported, I'm taking a look at it but I haven't been able to solve it yet

File: packages/x-components/src/plugins/index.ts [hunk 2]
```diff
@@ -1,6 +1,5 @@
 export * from './devtools/devtools.plugin';
 export * from './x-bus';
-export * from './x-bus.types';
 export * from './x-plugin';
```
[the remark this answers is not in the document]
Comment (LuisMartinez15) [reply packages/x-components/src/plugins/index.ts:3]: Done, thanks!

**Review by tiborux — APPROVED** [review]

**Review by mnavarroespinosa — APPROVED** [review]
good job!! 🔝

**Pull request merged**